compiler/rustc_trait_selection/src/traits/specialize/mod.rs

```diff
@@ -389,8 +380,7 @@ fn report_conflicting_impls(
     used_to_be_allowed: Option<FutureCompatOverlapErrorKind>,
     sg: &mut specialization_graph::Graph,
 ) {
-    let impl_span =
-        tcx.sess.source_map().guess_head_span(tcx.span_of_impl(impl_def_id.to_def_id()).unwrap());
+    let impl_span = tcx.span_of_impl(impl_def_id.to_def_id()).unwrap();
```

Comment:
```suggestion
    let impl_span = tcx.def_span(impl_def_id);
```
The `Ok` branch of `span_of_impl` is a call to `def_span`.  This allows to remove the unwrap, one less ICE hazard.